mlflow/gateway/uc_function_utils.py

```diff
@@ -0,0 +1,309 @@
+# TODO: Move this in mlflow/gateway/utils/uc_functions.py
+
+import json
+import re
+from dataclasses import dataclass
+from io import StringIO
+from typing import TYPE_CHECKING, Any, Dict, List, Literal, Optional, Union
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.catalog import FunctionParameterInfo
+
+if TYPE_CHECKING:
+    from databricks.sdk import WorkspaceClient
+    from databricks.sdk.service.catalog import FunctionInfo
+    from databricks.sdk.service.sql import StatementParameterListItem
+
+
+def uc_type_to_json_schema_type(uc_type_json: Union[str, Dict[str, Any]]) -> Dict[str, Any]:
+    """
+    Converts the JSON representation of a Unity Catalog data type to the corresponding JSON schema
+    type. The conversion is lossy because we do not need to convert it back.
+    """
+    # See https://docs.databricks.com/en/sql/language-manual/sql-ref-datatypes.html
+    # The actual type name in type_json is different from the corresponding SQL type name.
+    mapping = {
+        "long": {"type": "integer"},
+        "binary": {"type": "string"},
+        "boolean": {"type": "boolean"},
+        "date": {"type": "string", "format": "date"},
+        "double": {"type": "number"},
+        "float": {"type": "number"},
+        "integer": {"type": "integer"},
+        "void": {"type": "null"},
+        "short": {"type": "integer"},
+        "string": {"type": "string"},
+        "timestamp": {"type": "string", "format": "date-time"},
+        "timestamp_ntz": {"type": "string", "format": "date-time"},
+        "byte": {"type": "integer"},
+    }
```

Comment:
Should we construct this mapping using `ColumnTypeName` instead and add a test case verifying that every type is accounted for in the map? Else, we'll miss new types as they're added.